simulator/func_sim/alu.h

```diff
@@ -10,12 +10,84 @@
 #include "multiplication.h"
 #include "traps/trap.h"
 
+#include <array>
 #include <infra/macro.h>
```

Comment:
Please follow out guidelines on header order: https://github.com/MIPT-ILab/mipt-mips/wiki/Code-style-guidelines#order-of-includes